src/directives/formly-field.js

```diff
@@ -92,14 +92,16 @@ function formlyField($http, $q, $compile, $templateCache, $interpolate, formlyCo
       $scope.formOptions = $scope.formOptions || {};
     }
 
-    function setFieldId() {
+    function setFieldIdAndName() {
       if (angular.isFunction(formlyConfig.extras.getFieldId)) {
         $scope.id = formlyConfig.extras.getFieldId($scope.options, $scope.model, $scope);
       } else {
         const formName = ($scope.form && $scope.form.$name) || $scope.formId;
         $scope.id = formlyUtil.getFieldId(formName, $scope.options, $scope.index);
       }
+      $scope.name = $scope.options.name || $scope.options.id;
```

Comment:
No worries! I'm no git expert either. But I think that a git revert may not be the best way... One second...